python/lsst/pipe/analysis/fakesAnalysis.py

```diff
@@ -0,0 +1,615 @@
+        The catalog that catalo1 is matched to.
+    raCol1 : `string`
+        The column name for the RA column in catalog1.
+    decCol1 : `string`
+        The column name for the Dec. column in catalog1.
+    raCol2 : `string`
+        The column name for the RA column in catalog2.
+    decCol2 : `string`
+        The column name for the Dec. column in catalog2.
+    matchRadius : `float`
+        Radius within which to match the nearest source, in degrees, defaults to 0.1 arcseconds (0.1/3600.0).
+
+    Returns
+    -------
+    catalog1Matched : `pandas.core.frame.DataFrame`
+        Catalog with only the rows that had a match in catalog2.
+    catalog2Matched : `pandas.core.frame.DataFrame`
+        Catalog with only the rows that matched catalog1.
+
+    Notes
+    -----
+    matchRadius is in degrees, the catalogs need to be in degrees rather than radians. Returns two shortened
+    catalogs, with the matched rows in the same order and objects without a match removed. Matches the first
+    catalog to the second, multiple objects from the first catalog can match the same object in the second.
+    """
+
+    dists, inds = match_lists(catalog1[raCol1], catalog1[decCol1], catalog2[raCol2], catalog2[decCol2],
+                              matchRadius)
+
+    ids = (inds != len(catalog2))
+    matchedInds = inds[ids]
+    catalog1Matched = catalog1[ids]
+    catalog2Matched = catalog2.iloc[matchedInds].copy()
+
+    return catalog1Matched, catalog2Matched
+
+
+def addNearestNeighbor(catalog, raCol, decCol):
+    """Add the distance to the nearest neighbour in the catalog.
+
+    Parameters
+    ----------
+    catalog : `pandas.core.frame.DataFrame`
+        Catalog to add the distance to the nearest neighbour to.
+    raCol : `string`
+        Column name for the RA column to be used.
+    decCol : `string`
+        Column name for the Declination column to be used.
+
+    Returns
+    -------
+    catalog : `pandas.core.frame.DataFrame`
+        Catalog with a column 'nearestNeighbor' containing the distance to the neareast neighbour.
+
+    Notes
+    -----
+    Distance added is in degrees.
+    """
+
+    dists, inds = match_lists(catalog[raCol], catalog[decCol], catalog[raCol], catalog[decCol],
+                              20.0 / 3600, numNei=2)
+
+    distNN = np.array([d[1] for d in dists])
+    catalog["nearestNeighbor"] = distNN
+
+    return catalog
+
+
+def sigmaMAD(xs, med):
+    """Calculate the sigma given by the median absolute deviation.
+
+    Parameters
+    ----------
+    xs : `np.array`
+        Values to calculate the median absolute deviation of.
+    med : `float`
+        Median of the values in xs.
+
+    Returns
+    -------
+    sigmaMad : `float`
+        The sigma derived from the median absolute deviation.
+    """
+
+    sigmaMad = 1.4826 * np.median(np.fabs(xs - med))
+
+    return sigmaMad
+
+
+def getPlotInfo(repoInfo):
+    """Parse the repoInfo into a dict of useful info for plots.
+
```

Comment:
This sent me down a rabbit hole that landed at `getRepoInfo` which is... possibly either weirdly highly specialized and/or possibly really low level and useful and should live somewhere deep in the butler? At any rate, I was sad to learn that `butler.get('repo_info', dataId=dataId)` is not a thing, and I guess we can just move on for now.